### PR TITLE
Fix failing assertions on tests

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs [hunk 1]
```diff
@@ -236,7 +236,7 @@ private void CollectDumpAndAbortTesthost()
             {
                 Action<string> logWarning = m => _logger.LogWarning(_context.SessionDataCollectionContext, m);
                 var dumpDirectory = GetDumpDirectory();
-                _processDumpUtility.StartHangBasedProcessDump(_testHostProcessId, dumpDirectory, _hangDumpType == HangDumpType.Full, _targetFramework, logWarning);
+                _processDumpUtility.StartHangBasedProcessDump(_testHostProcessId, dumpDirectory, _hangDumpType == HangDumpType.Full, _targetFramework!, logWarning);
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs:239]: In failing tests `_targetFramework` is null but we pass it to some mock that doesn't care. Real life scenario, the implementation of this method throws on null TFM. I could also keep the assertion and update the tests to not pass null.

This comment also applies to the other assertion below.

**Review by nohwnd — APPROVED** [review]
